Sources/Spezi/Dependencies/DependencyCollection.swift

```diff
@@ -19,6 +19,14 @@ public struct DependencyCollection: DependencyDeclaration {
     init(_ entries: AnyDependencyContext...) {
         self.init(entries)
     }
+    
+    /// Creates a ``DependencyCollection`` from a closure resulting in a single generic type conforming to the Spezi  ``Module``.
+    /// - Parameters:
+    ///   - type: The generic type resulting from the passed closure, has to conform to ``Module``.
+    ///   - singleEntry: Closure returning a dependency conforming to ``Module``, stored within the ``DependencyCollection``.
+    public init<Dependency: Module>(for type: Dependency.Type = Dependency.self, singleEntry: @escaping (() -> Dependency)) {
```

Comment:
Nice, I get why the addition is made, but I would suggest that we add a small motivating example to both newly implemented public functions either with an example or a short note where they can be used to other developers understand how they can be used, e.g., to implements like in Spezi LLM 👍

Reply:
Thanks for the suggestion, added a complete, coherent implementation example 👍